### PR TITLE
Enable to skip line including comments

### LINKED ISSUE #352: Allow for comments in http format target files
With the [http-format](https://github.com/tsenart/vegeta#http-format) to specify targets it would be useful to allow for comments, e.g. lines starting with "#" should be ignored.

This would allow for developing your target file more easily, and for documenting it in general.

### PR DESCRIPTION
**Pull request by bookun**
### Background

fix #352 

#### Checklist

- [x] Git commit messages conform to [community standards](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] Each Git commit represents meaningful milestones or atomic units of work.
- [x] Changed or added code is covered by appropriate tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -236,6 +236,14 @@ X-Account-ID: 99
 @/path/to/newthing.json
 ```
 
+###### FYI: Add comments to the targets
```
[the remark this answers is not in the document]
Comment (bookun) [reply README.md]: It mean "For your Information".

File: lib/targets.go [hunk 2]
```diff
@@ -269,6 +269,10 @@ func NewHTTPTargeter(src io.Reader, body []byte, hdr http.Header) Targeter {
 				return ErrNoTargets
 			}
 			line = strings.TrimSpace(sc.Text())
+
+			if strings.HasPrefix(line, "#") {
+				continue
+			}
 			if len(line) != 0 {
 				break
 			}
```
[the remark this answers is not in the document]
Comment (bookun) [reply lib/targets.go]: OK
Your code is better than mine.
I will fix this.

File: lib/targets_test.go [hunk 3]
```diff
@@ -273,6 +273,11 @@ func TestNewHTTPTargeter(t *testing.T) {
 		errors.New("bad header"): `
 			GET http://:6060
 			: 1234`,
+		errors.New("bad method"): `
+			? a comment test
```
[the remark this answers is not in the document]
Comment (bookun) [reply lib/targets_test.go]: In a targets file, if a user writes comments started with from '#',  its comments are treated as comments. However, if he writes comments started with from character except for '#' or HTTP method name, the Vegeta return error "bad method".

I apologize for my poor English.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bookun) [reply README.md]: I excluded FYI.

File: lib/targets_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsenart) [reply lib/targets_test.go]: This line starts with a question mark `?`.

File: README.md [hunk 4]
```diff
@@ -236,6 +236,14 @@ X-Account-ID: 99
 @/path/to/newthing.json
 ```
 
+###### Add comments to the targets
+Lines started with from '#' are ignored.
```
Comment (tsenart) [inline README.md]: Lines starting with `#` are ignored.

File: lib/targets_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bookun) [reply lib/targets_test.go]: Anything other than `#` was fine.

File: README.md (hunk 4 above)
Comment (bookun) [reply README.md]: I fixed it.
Thank you.

File: lib/targets_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsenart) [reply lib/targets_test.go]: I don't think this test is necessary. The one below should be enough.

[the remark this answers is not in the document]
Comment (bookun) [reply lib/targets_test.go]: OK. I cut this test pattern.

**Pull request merged**